src/tools/haxelib/Main.hx

```diff
@@ -1287,18 +1263,29 @@ class Main {
 	function doGit(libName: String, gitPath: String, ?branch : String, ?subDir: String, ?version:String) {
 		var rep = getRepository();
 		var proj = rep + Data.safe(libName);
-		var libPath = proj + "/git";
 
+		// find existing repo
```

Comment:
There seems to be quite a lot of duplication between this and `doHg`. It also seems inconsistent to make `update` a responsibility of the `Vcs`, while `install` isn't. Could you try to structure this better?